packages/@uppy/react/src/useUppyStateSelector.js

```diff
@@ -0,0 +1,9 @@
+import { useSyncExternalStore } from 'react'
+
+export default function useUppyStateSelector (uppy, selector) {
+  const subscribe = uppy.store.subscribe.bind(uppy.store)
```

Comment:
Will this cause issues? See https://react.dev/reference/react/useSyncExternalStore#my-subscribe-function-gets-called-after-every-re-render

Reply:
Also, I don't completely understand why the binding is necessary here, same goes for the bind call below.

Reply:
I can wrap it in `useCallback` to be sure! Without the `bind`, I got a runtime crash because `this` was `undefined` in `@uppy/default-store`.

Reply:
I think `bind` is indeed necessary unless the implementation is using arrow functions for its `subscribe` method, e.g.
```js
subscribe = () => {}
```

however default-store does not: https://github.com/transloadit/uppy/blob/e9f335cbe64981eb7a2724b88322e935d2371aa3/packages/%40uppy/store-default/src/index.js#L26

And I agree `useCallback` is necessary here